src/rotator_library/request_sanitizer.py

```
@@ -1,17 +1,78 @@
-    return payload
+
+    # Kilocode provider - remove unsupported parameters for free models
+    # Free models through Kilocode/OpenRouter often reject extra parameters
+    if normalized_model.startswith("kilocode/"):
+        model_without_prefix = normalized_model.split("/", 1)[1] if "/" in normalized_model else normalized_model
+
+        # Free models have stricter parameter requirements
+        if ":free" in model_without_prefix or model_without_prefix.startswith("z-ai/"):
+            for param in KILOCODE_UNSUPPORTED_PARAMS:
+                if param in payload:
+                    del payload[param]
```

Ruff (0.15.4)

[warning] Use `pop` instead of `key in dict` followed by `del dict[key]`

Replace `if` statement with `.pop(..., None)`

(RUF051)